src/lib/routing-table.js

```diff
@@ -70,29 +70,31 @@ class RoutingTable {
       return undefined
     }
 
-    let bestHop = null
-    let bestValue = null
-    let bestRoute = null
+    let bestHop
     routes.forEach((route, nextHop) => {
-      const value = route.amountAt(sourceAmount)
       // If we have a route but not a curve, pick a route at random
       // and get a remote quote. In the future we may refactor this
       // so that multiple next hop options can be returned, and all
       // can be asked for a quote, but for now, it's just the last.
-      if (value === undefined || !bestValue || value.gt(bestValue)) {
-        bestHop = nextHop
-        bestValue = value && value.toString()
-        bestRoute = route
-      }
+      bestHop = getBetterHop(bestHop, {
+        nextHop,
```

Comment:
Do all of `currentHop`, `otherHop`, `nextHop` and `bestHop` have the same data type? If not, then that's confusing.